src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1303,11 +1306,31 @@ private bool ShouldCache(long number)
             blockNumber ??= _headerStore.GetBlockNumber(blockHash);
             if (blockNumber is not null)
             {
-                block = _blockStore.Get(
-                    blockNumber.Value,
-                    blockHash,
-                    (options & BlockTreeLookupOptions.ExcludeTxHashes) != 0 ? RlpBehaviors.ExcludeHashes : RlpBehaviors.None,
-                    shouldCache: false);
+                if (OperatingSystem.IsWindows())
```

Comment:
Should we cache it in some static variable? Or maybe some generic over struct way so it gets hard compiled?

Reply:
Its returns [a true/false const from runtime](https://github.com/dotnet/runtime/blob/8e2655b932f5f5e184d289982bdc516576df0981/src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs#L313), so will already do that 
<img width="540" alt="image" src="https://github.com/NethermindEth/nethermind/assets/1142958/9765a817-a31d-479b-87e8-452a616cddea">

Reply:
`[NonVersionable]` attribute means it can be inlined across libaries so at the "ready to run" publish step if targeting a platform